ports/ao/vcpkg.json

```
@@ -0,0 +1,16 @@
+{
+  "name": "ao",
```

```suggestion
  "name": "libao",
```

It looks like the library is pretty universally called `libao` elsewhere [1], so we'd like to use that name here as well. You'll need to rename `ports/ao` -> `ports/libao`.

[1] https://repology.org/project/libao/versions

---

> It looks like the library is pretty universally called `libao` elsewhere [1]

It calls itself just `ao`. https://github.com/xiph/libao/blob/20dc8ed9fa4605f5c25e7496ede42e8ba6468225/ao.pc.in#L8-L9
but is inconsistent about it : https://github.com/xiph/libao/blob/20dc8ed9fa4605f5c25e7496ede42e8ba6468225/ao.pc.in#L1

personally: the lib prefix is just a habit of linux to avoid collision with a binary called the same name. 
I really prefer the name from pc since it directly indicates usage: e.g.
`pkg-config --exists ao`

---

I see:
1. https://github.com/xiph/libao
2. Website url is https://www.xiph.org/ao/
2. Official description: > Libao is a cross-platform audio library
3. Official source code tarball is [libao-1.2.0.tar.gz](http://downloads.xiph.org/releases/ao/libao-1.2.0.tar.gz)
4. pkg-config is `ao`
5. The vast majority of other linuxes use `libao` (except GUIX uses `ao` https://guix.gnu.org/packages/ao-1.2.2-5-g20dc8ed/)
6. Brew uses https://formulae.brew.sh/formula/libao

I agree and understand that linuxes are motivated to use the lib prefix to avoid conflicts, but I think it promotes portability the most to use the common package name. It is very unfortunate that it disagrees with pkg-config though.

---

what about a compromise ,project name libao ,generate libarary ao.lib on windows.

---

`vcpkg_configure_make` should do the right thing about output names. 

> I agree and understand that linuxes are motivated to use the lib prefix to avoid conflicts, but I think it promotes portability the most to use the common package name.

I just find `lib<portname>` always hard to parse in between the other `lib<name>` stuff. Especially if `<portname>` is short. If we use a vote system I would say it is 3(github|description|most-other-package-name):2(pkg-config|website) for `libao`.

Personally I think vcpkg should have a consistent setting here and following upstream pkg-config seems to be our best defined bet. Especially since there are ports doing it correctly in pkg-config e.g. `libpng`. (I mean if we go with the linux package name approach `zlib` should have been called `libz`)

---

to further add to this. vcpkg could just add both ports `ao` and `libao` and make one of them a stub.

---

let @ras0219-msft decide before taking action ;)